packages/ra-supabase-ui-materialui/src/guessers/CreateGuesser.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import type { ReactNode } from 'react';
+import { useAPISchema } from 'ra-supabase-core';
+import {
+    useResourceContext,
+    Loading,
+    CreateBase,
+    CreateView,
+    InferredElement,
+    editFieldTypes,
+} from 'react-admin';
+import type { CreateProps, CreateViewProps } from 'react-admin';
+import { capitalize, singularize } from 'inflection';
+
+import { inferElementFromType } from './inferElementFromType';
+
+export const CreateGuesser = (props: CreateProps & { enableLog?: boolean }) => {
+    const {
+        mutationOptions,
+        resource,
+        record,
+        transform,
+        redirect,
+        disableAuthentication,
+        ...rest
+    } = props;
+    return (
+        <CreateBase
+            resource={resource}
+            record={record}
+            redirect={redirect}
+            transform={transform}
+            mutationOptions={mutationOptions}
+            disableAuthentication={disableAuthentication}
+        >
+            <CreateGuesserView {...rest} />
+        </CreateBase>
+    );
+};
+
+export const CreateGuesserView = (
+    props: CreateViewProps & {
+        enableLog?: boolean;
+    }
+) => {
+    const { data: schema, error, isPending } = useAPISchema();
+    const resource = useResourceContext();
+    const [child, setChild] = React.useState<ReactNode>(null);
+    if (!resource) {
+        throw new Error('CreateGuesser must be used withing a ResourceContext');
+    }
+    const { enableLog = process.env.NODE_ENV === 'development', ...rest } =
+        props;
+
+    React.useEffect(() => {
+        if (isPending || error) {
+            return;
+        }
+        const resourceDefinition = schema.definitions?.[resource];
+        const requiredFields = resourceDefinition?.required || [];
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            throw new Error(
+                `The resource ${resource} is not defined in the API schema`
+            );
+        }
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            return;
+        }
```

Comment:
nitpick (non-blocking): this code seems redundant with the lines above (same applies to other guessers)